scout/server/blueprints/cases/views.py

```diff
@@ -847,48 +895,20 @@ def phenotypes_actions(institute_id, case_name):
             case=case_obj,
         )
 
-    elif action == "ADDGENE":
-        hgnc_symbol = None
-        for raw_symbol in request.form.getlist("genes"):
-            LOG.debug("raw gene: {}".format(raw_symbol))
-            # avoid empty lists
-            if raw_symbol:
-                # take the first nubmer before |, and remove any space.
-                try:
-                    hgnc_symbol_split = raw_symbol.split("|", 1)[0]
-                    hgnc_symbol = int(hgnc_symbol_split.replace(" ", ""))
-                except ValueError:
-                    flash(
-                        "Provided gene info could not be parsed! "
-                        "Please allow autocompletion to finish.",
-                        "warning",
-                    )
-            LOG.debug("Parsed HGNC symbol {}".format(hgnc_symbol))
-            store.update_dynamic_gene_list(
-                case_obj, hgnc_ids=[hgnc_symbol], add_only=True
-            )
+    if action == "DELETE":
+        for hpo_id in hpo_ids:
+            # DELETE a phenotype from the list
+            store.remove_phenotype(institute_obj, case_obj, user_obj, case_url, hpo_id)
+
+    if action == "ADDGENE":
+        hgnc_ids = parse_raw_gene_ids(request.form.getlist("genes"))
+        store.update_dynamic_gene_list(case_obj, hgnc_ids=list(hgnc_ids), add_only=True)
+
+    if action == "GENES":
+        hgnc_symbols = parse_raw_gene_symbols(request.form.getlist("genes"))
```

Comment:
Never repeat code 😸

Reply:
Also never repeat two scarily similar but different pieces of code next to each other. 😅 I think the second try-catch- flash was never actually used (no int conversion to try for strings) and if it were it would have shown an incorrect error message..